torch/_functorch/partitioners.py

```diff
@@ -712,68 +798,113 @@ def get_sample_rng_state(device):
         )
 
     fw_rng_state_outputs = []
-    for base_node, node_pair in recomputable_rng_ops_map.items():
+
+    last_fwd_input = next(reversed(fw_module.graph.find_nodes(op="placeholder")))
+    last_bwd_input = next(reversed(bw_module.graph.find_nodes(op="placeholder")))
+
+    devices = OrderedSet(
+        get_device(node_pair["fwd"]) for node_pair in recomputable_rng_ops_map.values()
+    )
+    devices.discard(torch.device("cpu"))
```

Comment:
there may be non-cuda accelerator backends to compile that rely on AOTAutograd/partitioning. Should we make sure that we only actually execute this codepath for rng ops where `device=='cuda:X'`?

Reply:
We do below `device.type == "cuda"`.